libraries/AP_Frsky_Telem/AP_Frsky_Telem.h

```diff
@@ -112,14 +120,37 @@ for FrSky SPort Passthrough
 #define ATTIANDRNG_PITCH_OFFSET     11
 #define ATTIANDRNG_RNGFND_OFFSET    21
 // for fair scheduler
-#define TIME_SLOT_MAX               11
+#define TIME_SLOT_MAX               12
+// for two way communication
+#define SPORT_MAVLITE_MSG_SIZE(LEN) ((uint8_t)(1 + ceilf((LEN-2)/5.0f))) // number of sport packets required to transport a message with LEN payload
+#define TELEMETRY_RX_BUFFER_SIZE    19  // 9 bytes (full packet), worst case 18 bytes with byte-stuffing (+1)
+#define SPORT_PACKET_BUFFER_SIZE    30*SPORT_MAVLITE_MSG_SIZE(MAVLITE_MAX_PAYLOAD_LEN)
+#define SPORT_PACKET_SIZE           9
+#define STUFF_MASK                  0x20
+#define SPORT_DATA_FRAME            0x10
+#define SPORT_UPLINK_FRAME          0x30
+#define SPORT_UPLINK_FRAME_RW       0x31
+#define SPORT_DOWNLINK_FRAME        0x32
+
+using namespace MAVLITE;
 
 class AP_Frsky_Telem : public AP_RCTelemetry {
 public:
     AP_Frsky_Telem(bool external_data=false);
 
     ~AP_Frsky_Telem();
 
+    typedef union{
+        struct PACKED
```

Comment:
don't need PACKED ?

Reply:
you mean PACKED at the union, don't think so, we had a similar problem with the union in fport but in that case what was missing was the PACKED attribute of the 2nd struct in the union